src/server/config/complete.js

```diff
@@ -34,12 +31,19 @@ async function getUnusedConfigKeys(
   rawSettings,
   configValues
 ) {
-  // transform deprecated settings
-  const transforms = [
-    transformDeprecations,
-    ...await Promise.all(plugins.map(({ spec }) => getTransform(spec)))
-  ];
-  const settings = transforms.reduce((a, c) => c(a), rawSettings);
+  // transform deprecated core settings
```

Comment:
The short version is I had a **really** difficult time digging through the code originally and couldn't tell if we were expected to set by reference or value.  Turns out it does both, probably by accident.

We could deeply deprecate by reference (plugin.b.c), but not shallowly (plugin.b).  The weird part is it's not exactly shallow because it's one level deep.

So this sets everything by value.  I can put it in a new PR if we want, np.